apps/sms/views/conversation/test/unit/recipients_test.js

```diff
@@ -143,6 +143,24 @@ suite('Recipients', function() {
       assertDeepEqual(recipient, fixture);
     });
 
+    test('recipients.add() parenthesis ', function(done) {
+        recipients.on('add', function(count, added) {
```

Comment:
> I am not sure if what I have written is write. Any chance you could point out anything I am doing wrong? Or explanation on this particular case?

First of all as I said above, you call ```recipients.add``` twice so recipients ```add``` event handler is supposed to be run twice - but as soon as you call ```done()``` your test will be marked as finished - so the second ```recipients.add``` operation is not even checked - I'd suggest you to open Dev Tools and debug - to see what is really happening here.

Second thing is that you don't even need this ```add``` event handler at all - see the example in the test below - it doesn't deal with event handlers, so code looks cleaner.

Third thing is that you can reuse ```isValid``` method to validate that number is marked as valid (this method is used in the test below as well). But I see that ```isValid``` is too naive and may return ```true``` for the ```questionable``` number, so I'd suggest expand it with two additional checks ```candidate.isQuestionable``` and ```candidate.isValid```. 

Basically you should follow the rule - make test, see that it fails without your changes in code and succeed with them. Hope it helps